src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimeCustomAttributeData.cs

```diff
@@ -1278,7 +1277,14 @@ internal static object[] GetCustomAttributes(RuntimeType type, RuntimeType caTyp
 
             while (type != (RuntimeType)typeof(object) && type != null)
             {
-                AddCustomAttributes(ref result, type.GetRuntimeModule(), type.MetadataToken, caType, mustBeInheritable, result);
+                if (type.IsGenericParameter)
```

Comment:
I'm not confident about this line.
If the type is a method type parameter, there may be generic context from the declaring type. 
If the type is a type type parameter, is it possible to have a generic context?

Should I change it to
```cs
                if (type.IsGenericMethodParameter)
                {
                    AddCustomAttributes(ref result, type.GetRuntimeModule(), type.MetadataToken, caType, type.DeclaringType?.GetGenericArguments(), null, mustBeInheritable, result);
                }
                else if (!type.IsGenericTypeParameter)
                {
                    AddCustomAttributes(ref result, type.GetRuntimeModule(), type.MetadataToken, caType, type.GetGenericArguments(), null, mustBeInheritable, result);
                }
```
?